source/extensions/tracers/xray/daemon.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package envoy.tracers.xray.daemon;
+
+// see https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html
+message Segment {
```

Comment:
Do you think we should add some validations? ie. Mark some strings as non-empty, etc. The segment document spec you link actually lays out maximum string lengths and such fairly well. Even if this is just for documentation purposes it's nice to have the reference here.

Or perhaps add comments inline as to what these represent and expected values.